src/geoip.cpp

```diff
@@ -18,6 +18,40 @@ GeoDB::~GeoDB()
     MMDB_close(&mmdb);
 }
 
+std::string GeoDB::getGeoLocString(const uint32_t ip_address) const {
+
+    int mmdb_error;
+    struct sockaddr_in sa;
+    memset(&sa, 0, sizeof(sa));
+    memcpy(&sa.sin_addr, &ip_address, sizeof(sa.sin_addr));
+    sa.sin_family = AF_INET;
+    MMDB_lookup_result_s lookup = MMDB_lookup_sockaddr(&mmdb, (sockaddr *const)&sa, &mmdb_error);
+    if (mmdb_error != MMDB_SUCCESS || !lookup.found_entry) {
+        return "Unknown";
+    }
+
+    return _getGeoLocString(&lookup);
+
+}
+
+std::string GeoDB::getGeoLocString(const uint8_t* ip_address) const {
+
+    int mmdb_error;
+    struct sockaddr_in6 sa;
+    memset(&sa, 0, sizeof(sa));
+    sa.sin6_addr.s6_addr[0] = ip_address[0];
+    sa.sin6_addr.s6_addr[1] = ip_address[1];
+    sa.sin6_addr.s6_addr[2] = ip_address[2];
+    sa.sin6_addr.s6_addr[3] = ip_address[3];
```

Comment:
Nice catch, yes I thought that this macro pointed to the 4 byte (long) version of the address, not the byte. But I see on both OSX and linux it points to the byte (and thank god they are the same)
I will fix that, and also add a test. I was lulled into thinking this worked properly because the existing test passed.